### ISSUE TITLE
Weird crash in toJSONData

### ISSUE THREAD

**Issue by stefanhp** [Hearst-DD/ObjectMapper#645]
Hi,

I have a strange situation where tests in my project pass when run individually but crash when they are run as part of the suite of tests.

I'm using the latest (as of today) version of ObjectMapper and my project is in Swift 3 for iOS 10. The problem occurs both with Xcode 8.0 and Xcode 8.1 beta.

The crash happens on `if JSONSerialization.isValidJSONObject(JSONObject)` with EXC_BAD_ACCESS. The `JSONObject` seems properly formatted according to the debugger.

I understand this piece of code is from Foundation, so not your responsibility. But I was wondering if you already ran into such problems and if you know of a workaround.

Thanks.

**Comment by devxoul** [comment]
Could you please provide us how to reproduce the bug?

**Comment by stefanhp** [comment]
Hi,

Unfortunately I tried to isolate this in a test project, but it would not crash outside the original project...

But I found this radar bug: [iOS 10 Beta 6 - Swift 3 - JSONSerialization EXC_BAD_ACCESS](http://www.openradar.me/27925788) and the linked [gist](https://gist.github.com/aputinski/74416727b4e1153b9cee3708f544d809).

This could explain it: depending on what happened in the runtime before, the call may or may not crash. Prior to calling ```isValidJSONObject``` ObjectMapper transforms the object into a dictionary of [String:Any] (```public func toJSON(_ object: N) -> [String: Any]```) and the object I'm transforming contains Strings.

Since this seems to be a known Foundation issue, I guess you cannot do much about it. If so, I let you close the bug.
